src/window.rs

```diff
@@ -1035,6 +1027,35 @@ unsafe impl raw_window_handle::HasRawWindowHandle for Window {
     }
 }
 
+/// The behavior of cursor grabbing.
+///
+/// To grab cursor [`Window::set_cursor_grab_mode`] should be used.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum CursorGrabMode {
+    /// No grabbing of the cursor is performed.
+    None,
+
+    /// There's no guarantee that the cursor will be hidden. You should hide it by yourself if you
+    /// want so.
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **macOS:** Not implemented.
+    /// - ** iOS / Android / Web:** Always returns an [`ExternalError::NotSupported`].
+    Confined,
+
+    /// The cursor is locked inside the window area to the certain position.
+    ///
+    /// There's no guarantee that the cursor will be hidden, you should h
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **X11 / Windows:** Not implemented.
```

Comment:
The X11 and Windows backends both always return `ExternalError::NotSupported` in this case. This line makes it seem like they *might* not do so.

Reply:
The problem is that on those platforms it can be implemented. But on `ios`, `android`, etc won't ever have something like that.

Though I think we can also say that they always report.

Reply:
That's probably an important distincion, yeah.

Reply:
If it's the case that we *may* support this mode on X11 and Windows at some point, then I think it's a good idea to note that they return `ExternalError::NotSupported` for now, but may not do so in the future.

Reply:
Something like `Not implemented. Always return [ExternalError::NotSupported] for now.` ?

Reply:
Slap an "s" at the end of "return", and then that should suffice.

Reply:
Done.